cmd/xl-v1-object.go

```diff
@@ -339,8 +339,11 @@ func rename(disks []StorageAPI, srcBucket, srcEntry, dstBucket, dstEntry string,
 		undoRename(disks, srcBucket, srcEntry, dstBucket, dstEntry, isPart, errs)
 		return traceError(errXLWriteQuorum)
 	}
-	// Return on first error, also undo any partially successful rename operations.
-	return reduceErrs(errs, objectOpIgnoredErrs)
+	// Return on quorum error.
+	if reducedErr := reduceWriteQuorumErrs(errs, objectOpIgnoredErrs, quorum); reducedErr != nil {
```

Comment:
you could do ``return reduceWriteQuorumErrs(...)``. reduceWriteQuorumErrs handles all error  values alike, including nil.